packages/editor-ui/src/components/ParameterInput.vue

```diff
@@ -813,16 +814,25 @@ function valueChanged(value: NodeParameterValueType | {} | Date) {
 	if (remoteParameterOptionsLoading.value) {
 		return;
 	}
-	// Only update the value if it has changed
+
 	const oldValue = get(node.value, props.path);
+
 	if (oldValue !== undefined && oldValue === value) {
+		// Only update the value if it has changed
 		return;
 	}
 
+	if (!oldValue && oldValue !== undefined && isStringWithExpressionSyntax(value)) {
+		// if empty old value and updated value has an expression, add '=' prefix to swith to expression mode
```

Comment:
Why is the `!== undefined` check needed? typo: switch

Reply:
just precaution, for cases like with `assignments` and `conditions`, if old value would always be `undefined` it would be impossible to switch between `fixed` and `expression` modes